tests/integration/smcp-httpbin-test.sh

```diff
@@ -0,0 +1,77 @@
+#!/bin/bash
+
+
+# Copyright 2022 Red Hat, Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Exit immediately for non zero status
+set -e
+# Check unset variables
+set -u
+# Print commands
+set -x
+
+set -o pipefail
+
+WD=$(dirname "$0")
+WD=$(cd "$WD"; pwd)
+ROOT="$(git rev-parse --show-toplevel)"
+CR="${ROOT}/deploy/examples/maistra_v2_servicemeshcontrolplane_cr_minimal.yaml"
+BRANCH="${BRANCH:-maistra-2.3}"
+HTTPBIN="https://raw.githubusercontent.com/maistra/istio/${BRANCH}/samples/httpbin/httpbin.yaml"
+SLEEP="https://raw.githubusercontent.com/maistra/istio/${BRANCH}/samples/sleep/sleep.yaml"
+DPNS="default"
+
+create-control-plane() { # creates a SMCP with a mini CR
+    local CPNS="${CPNS:-istio-system}"
+    kubectl get ns "${DPNS}" >/dev/null 2>&1 || kubectl create namespace "${DPNS}"
+    kubectl get ns "${CPNS}" >/dev/null 2>&1 || kubectl create namespace "${CPNS}"
+
+    sed "s:#- bookinfo:- ${DPNS}:g" "${CR}" | kubectl apply -n "${CPNS}" -f -
+
+    kubectl wait --for condition=Ready -n "${CPNS}" smcp/minimal --timeout 300s
+    kubectl wait --for condition=Ready -n "${CPNS}" smmr/default --timeout 180s
+    kubectl get -n "${CPNS}" smcp -o wide
+}
+
+create-httpbin() { # creates a httpbin app and injects sidecar proxy
+    kubectl apply -n "${DPNS}" -f "${HTTPBIN}"
+    kubectl apply -n "${DPNS}" -f "${SLEEP}"
+    kubectl wait --for condition=Ready -n "${DPNS}" pod -l app=httpbin --timeout 300s
+    kubectl wait --for condition=Ready -n "${DPNS}" pod -l app=sleep --timeout 180s
+}
+
+check-httpbin() {
+    # checks if a httpbin pod has two containers
+    echo "Check httpbin pod status:"
+    kubectl get -n "${DPNS}" pod
+    kubectl get -n "${DPNS}" pod -l app=httpbin \
+        -o=jsonpath='{range .items[*]}{.metadata.name}{" "}{.status.containerStatuses[*].ready}{"\n"}{end}' \
+        | grep 'true true'
+
+    echo "Check if a request is routed through the proxy:"
+    # maistra prow kube-dns Could not resolve host: httpbin
+    # use service clusterIP in tests
+    local SVCIP
+    SVCIP="$(kubectl -n default get service httpbin \
+        -o jsonpath='{.spec.clusterIP}')"
+    kubectl exec -n "${DPNS}" \
+        "$(kubectl get -n ${DPNS} pod -l app=sleep -o jsonpath={.items..metadata.name})" \
+        -c sleep -- curl -sS "http://${SVCIP}:8000/headers" \
```

Comment:
This can be simplified to:
```suggestion
    kubectl exec -n "${DPNS}" deploy/sleep -c sleep -- curl -sS "http://${SVCIP}:8000/headers" \
```